components/claim-mgt/org.wso2.carbon.identity.claim.metadata.mgt/src/main/java/org/wso2/carbon/identity/claim/metadata/mgt/dao/CacheBackedLocalClaimDAO.java

```diff
@@ -115,4 +116,17 @@ public void removeClaimMappingAttributes(int tenantId, String userstoreDomain) t
         localClaimDAO.deleteClaimMappingAttributes(tenantId, userstoreDomain);
         localClaimInvalidationCache.clearCacheEntry(tenantId, tenantId);
     }
+
+    /**
+     * Fetch mapped external claims of a local claim.
+     *
+     * @param tenantId        Tenant Id.
+     * @param localClaimURI   URI of local claim.
+     * @throws ClaimMetadataException If an error occurred while getting mapped external claim for local claim.
+     */
+    public List<Claim> fetchMappedExternalClaims(String localClaimURI, int tenantId)
+            throws ClaimMetadataException {
+
```

Comment:
Need to cache this result. Otherwise, every local claim's mapped external role retrieval hits the DB

Reply:
Added with https://github.com/wso2/carbon-identity-framework/pull/4442/commits/ee9f7b1799c158e786d1101ed166dd2b36e67aec